data_structures/heap/min_heap.ts

```diff
@@ -32,8 +32,8 @@ export class MinHeap<T> extends Heap<T> {
     leftChildIndex: number,
     rightChildIndex: number
   ): number {
-    return (this.heap[leftChildIndex] || -Infinity) <
-      (this.heap[rightChildIndex] || -Infinity)
+    return this.heap[leftChildIndex] <
+      (this.heap[rightChildIndex] || Infinity)
```

Comment:
Two changes here:

1. Remove the `||` of the left hand side. The left child should never be undefined on any code path, so its unnecessary. Non-functional change
2. `-Infinity` -> `Infinity`. This actually fixes the bugs. This makes sure if the right hand side is `undefined`, we always select the left child.